### PR TITLE
[AIX] Compile with `-pthread` if LLVM_ENABLE_THREADS

### PR DESCRIPTION
**Pull request by bzEq**
LTO may be multithreaded. Specify this flag to ensure thread safety on AIX.

### CODE REVIEW

File: llvm/CMakeLists.txt [hunk 1]
```diff
@@ -1150,6 +1150,11 @@ if (UNIX AND ${CMAKE_SYSTEM_NAME} MATCHES "AIX")
           add_compile_definitions(_XOPEN_SOURCE=700)
           add_compile_definitions(_LARGE_FILE_API)
 
+  # Make functions, like errno, thread safe.
+  if (LLVM_ENABLE_THREADS)
+    add_compile_definitions(_THREAD_SAFE)
```
Comment (hubert-reinterpretcast) [inline llvm/CMakeLists.txt:1155]: Why do we need to define this macro manually? Would using `-pthread` make more sense?

Comment (bzEq) [reply llvm/CMakeLists.txt:1155]: Ok, I see, `-pthread` implies `_THREAD_SAFE` on AIX. Thanks for pointing it out.

**Comment by bzEq** [comment]
Close it as this macro is implied by `-pthread`.

**Comment by bzEq** [comment]
I checked `compile_commands.json` and looks `-pthread` is not specified when `LLVM_ENABLE_THREADS`. Re-open it for further discussion.

**Comment by bzEq** [comment]
Still close it after search on AIX's headers, `_XOPEN_SOURCE==700` has implied `_THREAD_SAFE`.

**Pull request closed without merging**